### ISSUE TITLE
Add tags to search differently - e.g. @folder:work, @after:2d, @before:1m

### ISSUE THREAD

**Issue by marvingreenberg** [xbrowsersync/app#142]
**Is your feature request related to a problem?**
I think it is related to the tags losing dates on the initial sync, or something.  I've just synced and hitting return on the sync search is returning bookmarks from years ago.  But, the capability is still probably useful since if I want to find some bookmarks from the summer saying something like @before:3m (before 3 months ago) or @before:20190901 will be useful.

**Tell us about your idea**
This is related to #16, but it is unclear to me how THAT would be implemented, and seems to need a lot of UI.
I'm just suggesting something I _think_ is simple.  If xb-sync can has access to the folder structure then typing ```@folder:work``` would find bookmarks in ```work/cloud``` and ```work/java``` (and ```work/```) while ```@folder:cloud``` would just show those in ```work/cloud``` (unless there were another dir like ```home/cloud```).  Personally I'd keep it that simple since its easy to explain.  Each directory essentially adds a tag.  Other useful meta-searches are date based (```@before:201909 @after:201906```) and URL based ```@site:msdn.microsoft.com``` (but, sorry if you have to read MS docs).

**Comment by nero120** [comment]
Thanks for the suggestion @marvingreenberg, meta searches is a great idea and would definitely like to implement this kind of thing in a future release.

Just for info, currently xBrowserSync does not store dates with bookmarks so querying by date wouldn't be possible at this time.

**Comment by marvingreenberg** [comment]
Doesn't the existing... Oh, it says sorted by most recent, but that's just the order (or some order) of addition.  It seems like it could add the dates as things are synced, perhaps.  It wouldn't have to be super accurate just to the day or so, which presumably sync date would provide.

**Comment by nero120** [comment]
Yes it's the order of addition, bookmarks are shown by default in search by order of id descending, which implies most recently added are displayed first. Once you type a query they are displayed by relevancy. It was a design choice not to include dates as the Web Extension APIs include no ability to set/update or query dates, and the space required to store all of these dates would limit the amount of free users the service could support. I don't think the case has changed so I have no plans to include syncing bookmark dates at this time.

**Comment by marvingreenberg** [comment]
Fair enough.  But, not meaning to be argumentative, you could store year and day as YYddd. Like 19350 for today (maybe). Which would take 2 bytes (per bookmark, at least for the next 40 years). Maybe 5%? Stripping the most common scheme https:// would get you back 8 bytes (probably you do that already).

Personally I'm disorganized.  When I'm trying to find something I bookmarked I want more ways to find it and being able to search more than one way (I'm sure it was a bookmark I did last September...) Is useful FOR ME.

That said, all this searching by tags is (probably) a small use case, and you have to prioritize things you think are important.  If I really care, I can make a pull request or just implement it for myself, though it would be more work (for me) to do it for the Apps.  Thanks.

**Comment by nero120** [comment]
Thanks for the feedback @marvingreenberg, it's definitely appreciated. I'm sure it will be implemented at some point, as you say though have to prioritise things aggressively due to limited spare time.
